deployment/ccip/changeset/testhelpers/test_helpers.go

```diff
@@ -1835,6 +1885,15 @@ func Transfer(
 		t.Errorf("unsupported source chain: %v", family)
 	}
 
+	onRampAddr, err := state.GetOnRampAddressBytes(sourceChain)
+	require.NoError(t, err)
+	// Ensure CCIPMessageSent event filter is registered
+	// Sending message too early could result in LogPoller missing the send event
+	err = WaitForEventFilterRegistration(t, env.Offchain, sourceChain, consts.EventNameCCIPMessageSent, onRampAddr)
```

Comment:
Is there a way to do this categorically? Are we waiting for the "discovery" process to find contract addresses or is this some general node startup race?

Reply:
The ChainReader should have a Ready function since it implements `services.StateMachine` and uses the normal `StartOnce`. I don't think I see it being checked during initialization in the Plugin though [here](https://github.com/smartcontractkit/chainlink/blob/759a9d93cd5d92efbede8e43d2e44877b32f577a/core/capabilities/ccip/oraclecreator/plugin.go#L491). I'm not too familiar with the code in case it's happening somewhere else